### PR TITLE
mupdf: update to 1.19.1

### PR DESCRIPTION
**Pull request by markmentovai**
This also switches the libmupdf build from a static library (libmupdf.a)
to a shared library (libmupdf.dylib).

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 12.3.1 21E258 arm64
Xcode 13.3.1 13E500a

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: graphics/mupdf/Portfile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 PortSystem          1.0
 
 name                mupdf
-version             1.17.0
+version             1.19.1
 revision            1
```
Comment (reneeotten) [inline graphics/mupdf/Portfile:7]: the "revision" field should be reset to zero when updating the version

File: graphics/mupdf/Portfile [hunk 2]
```diff
@@ -58,13 +59,15 @@ build.args-append   CC=${configure.cc} \
                     HAVE_X11=no \
                     USE_SYSTEM_LIBS=yes \
                     build=release \
-                    verbose=yes
+                    verbose=yes \
+                    shared
```
Comment (reneeotten) [inline graphics/mupdf/Portfile:63]: does making this change will require a revision bump for all the dependents of this port or not?

File: graphics/mupdf/Portfile (hunk 1 above)
Comment (markmentovai) [reply graphics/mupdf/Portfile]: Fixed.

File: graphics/mupdf/Portfile (hunk 2 above)
Comment (markmentovai) [reply graphics/mupdf/Portfile:63]: Yes, there’s only one binary dependency that needed both a revision bump and a build fix. Done, thanks.

Comment (reneeotten) [reply graphics/mupdf/Portfile:63]: can you please make that change to `zathura-plugin-pdf-mupdf` in a separate commit in this PR?

Comment (markmentovai) [reply graphics/mupdf/Portfile:63]: Sure, updated.

**Comment by reneeotten** [comment]
@harciga any comments, if not I'll merge this tomorrow after the typical 72h

**Pull request merged**